### PR TITLE
increase width for ecr-viewer and improve responsiveness

### LINKED ISSUE #2180: Responsive Viewer | eCR Viewer
## User Story

As somebody looking at the eCR Viewer, I want to be able to use the eCR Viewer at different browser sizes, so that I can more easily view complex tables and use the viewer even when I have multiple windows open on my desktop

## Acceptance Criteria

- [ ] The width of the left navigation menu never changes
- [ ] The minimum width of the eCR Viewer content (to the right of the nav menu) is set to `40 rem` (not confident on this, will need to see how this looks in design review)
- [ ] The maximum width of the eCR Viewer is set to `80 rem` (not confident on this, will need to see how this looks in design review)
- [ ] The content is responsive to browser resizing between 40 rem and 80 rem, smoothly changing width relative to the browser size (i.e. no weird jumps in placement or drastic padding changes)

## Scenarios

Scenario 1:
- I start with the Viewer open in a browser as large as possible on a large desktop computer
- The maximum width is at 80 rem
- I then make my browser gradually narrower
- The Viewer content gradually and smoothly gets smaller until it reaches the minimum width of 40 rem
- The width of the left side navigation has stayed the same the whole time

## Additional context

It's important to make the eCR Viewer more responsive because we've seen our users most frequently have many windows open at the same time to reference across systems. We want to make sure the Viewer still has a good experience with narrow browser sizes, and also can better expand for users that want to view in full screen on a large desktop to see table data more spread out.

For now we're going to avoid thinking about super narrow screens (mobile) because our users won't be able to use it on mobile devices, and are unlikely to make their browser windows that narrow

### PR DESCRIPTION
**Pull request by BobanL**
# PULL REQUEST

## Summary
- Change width of ecr viewer to 80rem and minimum 50rem
- Move css utility classes into their own file

## Related Issue
Fixes #2180 

## Additional Information
new:
[min screen responsive.pdf](https://github.com/user-attachments/files/16499868/min.screen.responsive.pdf)
[full screen responsive.pdf](https://github.com/user-attachments/files/16499870/full.screen.responsive.pdf)
old:
[old.pdf](https://github.com/user-attachments/files/16499873/old.pdf)

Can also be reviewed via the [design review script](https://github.com/CDCgov/phdi/tree/main/containers/ecr-viewer/design-review)
## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Comment by sarahtress** [comment]
@BobanL got the design script working! a few requests:

- could we try 40rem for min width overall? 50 seems a bit too big
- when I make my browser smaller than 40, I should be able to scroll right and left in the browser to see the full doc, but I'm not able to scroll right or left, which makes it impossible to see the full content. can we make sure you're able to scroll right and left when narrower than the min width?
- all of the tables are looking good on resizing, except the Lab result tables are only resizing on Value (in screenshot below). could we ensure the columns all resize more evenly and also get the headers to always be on one line (like Ref Range and Lab Comment) if possible?

Screenshot 2024-08-05 at 2 51 07 PM

**Comment by BobanL** [comment]
Ready for re-review @sarahtress!

**Comment by sarahtress** [comment]
@BobanL sorry seeing some different funky stuff now! see screenshots below:

Screenshot 2024-08-06 at 2 29 00 PM
Screenshot 2024-08-06 at 2 28 44 PM
Screenshot 2024-08-06 at 2 27 35 PM
Screenshot 2024-08-06 at 2 26 58 PM

**Comment by BobanL** [comment]
> @BobanL sorry seeing some different funky stuff now! see screenshots below:
> 
> something weird going on with Facility Name here: Screenshot 2024-08-06 at 2 29 00 PM
> 
> these 2 tables aren't responsive with the rest: Screenshot 2024-08-06 at 2 28 44 PM
> 
> there is a gap of white space on the right with really wide browsers and really narrow browsers and the content isn't centered: Screenshot 2024-08-06 at 2 27 35 PM Screenshot 2024-08-06 at 2 26 58 PM

Fixed the facility name issue. As for the rest, many of those things were happening yesterday evening, but it doesn't look like that for me locally :? If it's still happening after that latest change I'd love to pair to see whats going. 

https://github.com/user-attachments/assets/921e287d-8643-4e83-92f8-5366015bd2b4

**Comment by sarahtress** [comment]
> > @BobanL sorry seeing some different funky stuff now! see screenshots below:
> > something weird going on with Facility Name here: Screenshot 2024-08-06 at 2 29 00 PM
> > these 2 tables aren't responsive with the rest: Screenshot 2024-08-06 at 2 28 44 PM
> > there is a gap of white space on the right with really wide browsers and really narrow browsers and the content isn't centered: Screenshot 2024-08-06 at 2 27 35 PM Screenshot 2024-08-06 at 2 26 58 PM
> 
> Fixed the facility name issue. As for the rest, many of those things were happening yesterday evening, but it doesn't look like that for me locally :? If it's still happening after that latest change I'd love to pair to see whats going.
> 
>  Screenshare.-.2024-08-06.4_06_20.PM.1.mp4

weird okay let me try again!

**Comment by sarahtress** [comment]
@BobanL ah yup looks good to me now! no more comments on my end

**Review by angelathe — COMMENTED** [review]
Generally, LGTM! Just one question (and it may not even be in the scope of this ticket). 

When testing, I noticed that changing the browser widths can affect the current place in the eCR Viewer that the user is looking at (see video) -- this happens especially when multiple accordions are open at the same time (i.e., lab results). Is this something that we should be concerned about? 

Uploading Untitled.mov…

**Comment by BobanL** [comment]
> Generally, LGTM! Just one question (and it may not even be in the scope of this ticket).
> 
> When testing, I noticed that changing the browser widths can affect the current place in the eCR Viewer that the user is looking at (see video) -- this happens especially when multiple accordions are open at the same time (i.e., lab results). Is this something that we should be concerned about?
> 
>  Untitled.mov

Looking at other sites it seems like similar things happen. Now I'm not sure if thats an issue for all of these sites though. 

image
image

image
image

**Review by angelathe — APPROVED** [review]
